internal/controllers/nodequotaconfig_controller.go

```diff
@@ -68,47 +72,64 @@ func (r *NodeQuotaConfigReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&danav1alpha1.NodeQuotaConfig{}).
 		Watches(
-			&source.Kind{Type: &corev1.Node{}},
+			&corev1.Node{},
 			handler.EnqueueRequestsFromMapFunc(r.requestConfigReconcile),
 		).
 		Complete(r)
 }
 
-func (r *NodeQuotaConfigReconciler) CalculateRootSubnamespaces(ctx context.Context, config danav1alpha1.NodeQuotaConfig) error {
+// CalculateRootSubnamespaces calculates the resource allocation for the root subnamespaces based on the provided NodeQuotaConfig.
+// It takes a context, the NodeQuotaConfig to reconcile, and a logger for logging informational messages.
+// It returns an error (if any occurred) during the calculation.
+func (r *NodeQuotaConfigReconciler) CalculateRootSubnamespaces(ctx context.Context, config danav1alpha1.NodeQuotaConfig, logger logr.Logger) error {
 	oldConfigStatus := config.Status.DeepCopy()
-	reservedResources := []danav1alpha1.ReservedResources{}
 	for _, rootSubnamespace := range config.Spec.Roots {
+		logger.Info(fmt.Sprintf("Starting to calculate RootSubnamespace %s", rootSubnamespace.RootNamespace))
 		rootResources := v1.ResourceList{}
+		processedSecondaryRoots := []danav1.Subnamespace{}
 		for _, secondaryRoot := range rootSubnamespace.SecondaryRoots {
-			err, secondaryRootResources := utils.ProcessSecondaryRoot(ctx, r.Client, secondaryRoot, &config, rootSubnamespace.RootNamespace)
+			logger.Info(fmt.Sprintf("Starting to calculate Secondary root %s", secondaryRoot.Name))
+			err, secondaryRootSns := utils.ProcessSecondaryRoot(ctx, r.Client, secondaryRoot, &config, rootSubnamespace.RootNamespace, logger)
 			if err != nil {
 				return err
 			}
-			rootResources = utils.MergeTwoResourceList(secondaryRootResources, rootResources)
+			processedSecondaryRoots = append(processedSecondaryRoots, secondaryRootSns)
+			rootResources = utils.MergeTwoResourceList(secondaryRootSns.Spec.ResourceQuotaSpec.Hard, rootResources)
 		}
 		rootRQ, err := utils.GetRootQuota(r.Client, ctx, rootSubnamespace.RootNamespace)
 		if err != nil {
+			logger.Error(err, fmt.Sprintf("Error getting the %s resourceQuota", rootSubnamespace.RootNamespace))
 			return err
 		} else {
-			rootRQ.Spec.Hard = utils.FillterUncontrolledResources(rootResources, config.Spec.ControlledResources)
+			rootRQ.Spec.Hard = rootResources
+			logger.Info(fmt.Sprintf("Updating RootSubnamespace %s with new resources", rootSubnamespace.RootNamespace))
 			if err := r.Update(ctx, &rootRQ); err != nil {
 				return err
 			}
```

Comment:
Put the update of the root namespace wallet and the update of the secondary roots in different functions